### PR TITLE
refactor(cardinal): move /query/debug/state from plugins to http handlers

### PR DESCRIPTION
**Pull request by jerargus**
Closes: WORLD-996

## Overview

Move the debug state handler out of plugins and back over to an HTTP handler.
Note, the cardinal/server/handler/debug.go was jus moved, and there are no actual logic changes.

The debug_test file was changed to 1) hit the HTTP endpoint instead of using the query directly and 2) use the structs that already exist in the test suite.

## Testing and Verifying

Existing tests were modified to hit the HTTP endpoint instead of using the query.

### CODE REVIEW

File: cardinal/server/handler/debug.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	ID         types.EntityID             `json:"id"`
+	Components map[string]json.RawMessage `json:"components" swaggertype:"object"`
+}
+
+type DebugStateResponse []*debugStateElement
+
+// GetDebugState godoc
+//
+//	@Summary		Get information on all entities and components in world-engine
+//	@Description	Displays the entire game state.
```
Comment (smsunarto) [inline cardinal/server/handler/debug.go:25]: nit: formatting seems borked here

File: cardinal/server/handler/debug.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+//	@Produce		application/json
+//	@Success		200	{object}	DebugStateResponse
```
Comment (smsunarto) [inline cardinal/server/handler/debug.go:27]: nit: we should probably annotate possible internal server error (Error 500)

File: cardinal/server/handler/debug.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+type debugStateElement struct {
+	ID         types.EntityID             `json:"id"`
+	Components map[string]json.RawMessage `json:"components" swaggertype:"object"`
+}
+
+type DebugStateResponse []*debugStateElement
+
+// GetDebugState godoc
+//
+//	@Summary		Get information on all entities and components in world-engine
```
Comment (smsunarto) [inline cardinal/server/handler/debug.go:24]: nit: `Get all entity and component state in Cardinal`

File: cardinal/server/handler/debug.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+)
+
+type DebugStateRequest struct{}
+
+type debugStateElement struct {
+	ID         types.EntityID             `json:"id"`
+	Components map[string]json.RawMessage `json:"components" swaggertype:"object"`
+}
+
+type DebugStateResponse []*debugStateElement
```
Comment (smsunarto) [inline cardinal/server/handler/debug.go:20]: is there a reason why this is a pointer to struct? without pointer seems cleaner.

**Review by smsunarto — APPROVED** [review]
left some optional nits

File: cardinal/server/handler/debug.go (hunk 1 above)
Comment (jerargus) [reply cardinal/server/handler/debug.go]: done

File: cardinal/server/handler/debug.go (hunk 2 above)
Comment (jerargus) [reply cardinal/server/handler/debug.go]: done

File: cardinal/server/handler/debug.go (hunk 4 above)
Comment (jerargus) [reply cardinal/server/handler/debug.go]: No reason. It was copied from the plugin code. Changed to value.

**Review by technicallyty — APPROVED** [review]

**Pull request merged**